influxdb3_client/src/lib.rs

```diff
@@ -0,0 +1,382 @@
+    ) -> QueryRequestBuilder<'_> {
+        QueryRequestBuilder {
+            client: self,
+            db: db.into(),
+            query: query.into(),
+            format: None,
+        }
+    }
+}
+
+/// The body of the request to the `/api/v3/write_lp` API
+// TODO - this should re-use the type defined in the server code.
+#[derive(Debug, Serialize)]
+struct WriteParams<'a> {
+    db: &'a str,
+    precision: Option<Precision>,
+    accept_partial: Option<bool>,
+}
+
+impl<'a, B> From<&'a WriteRequestBuilder<'a, B>> for WriteParams<'a> {
+    fn from(builder: &'a WriteRequestBuilder<'a, B>) -> Self {
+        Self {
+            db: &builder.db,
+            precision: builder.precision,
+            accept_partial: builder.accept_partial,
+        }
+    }
+}
+
+/// Time series precision
+// TODO - this should re-use type from server code.
+#[derive(Debug, Copy, Clone, Serialize)]
+#[serde(rename_all = "snake_case")]
+pub enum Precision {
+    Second,
+    Milli,
+    Micro,
+    Nano,
+}
+
+/// Builder type for composing a request to `/api/v3/write_lp`
+///
+/// Produced by [`Client::api_v3_write_lp`]
+#[derive(Debug)]
+pub struct WriteRequestBuilder<'c, B> {
+    client: &'c Client,
+    db: String,
+    precision: Option<Precision>,
+    accept_partial: Option<bool>,
+    body: B,
+}
+
+impl<'c, B> WriteRequestBuilder<'c, B> {
+    /// Set the precision
+    pub fn precision(mut self, set_to: Precision) -> Self {
+        self.precision = Some(set_to);
+        self
+    }
+
+    /// Set the `accept_partial` parameter
+    pub fn accept_partial(mut self, set_to: bool) -> Self {
+        self.accept_partial = Some(set_to);
+        self
+    }
+}
+
+impl<'c> WriteRequestBuilder<'c, NoBody> {
+    /// Set the body of the request to the `/api/v3/write_lp` API
+    ///
+    /// This essentially wraps `reqwest`'s [`body`][reqwest::RequestBuilder::body]
+    /// method, and puts the responsibility on the caller for now.
+    pub fn body<T: Into<Body>>(self, body: T) -> WriteRequestBuilder<'c, Body> {
+        WriteRequestBuilder {
+            client: self.client,
+            db: self.db,
+            precision: self.precision,
+            accept_partial: self.accept_partial,
+            body: body.into(),
+        }
+    }
+}
+
+impl<'c> WriteRequestBuilder<'c, Body> {
+    /// Send the request to the server
+    pub async fn send(self) -> Result<()> {
+        let url = self.client.base_url.join("/api/v3/write_lp")?;
+        let req = WriteParams::from(&self);
+        let mut b = self.client.http_client.post(url).query(&req);
+        if let Some(token) = &self.client.auth_header {
+            b = b.bearer_auth(token.expose_secret());
+        }
+        let resp = b.body(self.body).send().await.map_err(Error::WriteLpSend)?;
+        let status = resp.status();
+        let content = resp.bytes().await.map_err(Error::Bytes)?;
+        match status {
+            // TODO - handle the OK response content, return to caller, etc.
```

Comment:
I think it should be an empty body. I'm not sure why we left it as just `{}`. @pauldix were we planning on returning more on a successful response in the future for the write_lp endpoint? I don't see a reason too, but I'm probably missing context.

Reply:
Yeah, I was wondering if there would be stats returned, e.g., number of lines inserted, or in the case of `accept_partial = true`, the rejected lines, etc.